runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiPlugin.java

```diff
@@ -245,6 +248,18 @@ private void onMenuOptionClicked(MenuOptionClicked ev)
 					location = WorldPoint.fromScene(client, ev.getActionParam(), ev.getWidgetId(), client.getPlane());
 					break;
 				}
+				case SPELL_CAST_ON_WIDGET:
+					Widget w = getWidget(ev.getWidgetId(), ev.getActionParam());
+
+					if (w.getType() == WidgetType.GRAPHIC && w.getItemId() != -1)
+					{
+						type = "item";
+						id = itemManager.canonicalize(w.getItemId());
+						name = itemManager.getItemComposition(id).getName();
+						location = null;
+						break;
+					}
+					// fallthrough
```

Comment:
Is this correct? There are definitely non-item graphics.

Reply:
Thats why I check `itemID != -1`

Reply:
Right, but if not it will fallthrough and log a warning about it

Reply:
It shouldn't have gotten a menuop if its not an item, so logging a warning is appropriate.